client/src/fabric/FabricRuntimeManager.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+import { FabricRuntime } from './FabricRuntime';
+import { FabricRuntimeRegistry } from './FabricRuntimeRegistry';
+import { FabricRuntimeRegistryEntry } from './FabricRuntimeRegistryEntry';
+import { FabricConnectionRegistry } from './FabricConnectionRegistry';
+import { FabricConnectionRegistryEntry } from './FabricConnectionRegistryEntry';
+
+export class FabricRuntimeManager {
+
+    public static instance(): FabricRuntimeManager {
+        return this._instance;
+    }
+
+    private static _instance = new FabricRuntimeManager();
+
+    private connectionRegistry: FabricConnectionRegistry = FabricConnectionRegistry.instance();
+    private runtimeRegistry: FabricRuntimeRegistry = FabricRuntimeRegistry.instance();
+    private runtimes: Map<string, FabricRuntime> = new Map<string, FabricRuntime>();
+
+    private constructor() {
```

Comment:
Yes - this is a singleton pattern - make the constructor private, so that only the class can construct itself - which it does further up for `_instance`.